### PR TITLE
like redux

### PR DESCRIPTION
**Pull request by otuffaha**

### CODE REVIEW

File: src/reducers/RecipeReducer.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ const initialState = {
   recipes: [],
   recipeDetails: {},
   recipeRecs: [],
+  likedRecipes: [],
+  recipeToLike: {},
```
Comment (seanyles) [inline src/reducers/RecipeReducer.js]: It doesn't make sense to have recipetoLike or unLike. The back end just updates the recipe by id to be liked or unliked. I believe the use case is in the details page, so probably have the respective actions with fetch and receive the recipeDetails of the recipe in question. @mavalos1 please provide your thoughts

Comment (otuffaha) [reply src/reducers/RecipeReducer.js]: We should keep the action types to differentiate the action clearly, but i eliminated those two properties and pointed them to the recipeDetails property instead. Please let me know if that is cool with you.

**Review by seanyles — APPROVED** [review]

**Comment by seanyles** [comment]
merge if / when tested

File: src/services/RecipeService.js [hunk 2]
```diff
@@ -23,6 +23,21 @@ class RecipeService {
...
   };
+
+  getLikedRecipes = () => {
+    const route = `${routes.RECIPE}/${routes.LIKE}`;
+    return getHelper(route);
+  }
+
+  getRecipeToLike = (recipeId) => {
+    const route = `${routes.RECIPE}/${routes.LIKE}/${recipeId}`;
+    return getHelper(route);
```
Comment (seanyles) [inline src/services/RecipeService.js]: I believe the like route is POST, and the unlike route is DELETE. Look at the back end for the signature. They return collections I think

**Review by seanyles — CHANGES_REQUESTED** [review]
Edit services

File: src/services/RecipeService.js (hunk 2 above)
Comment (otuffaha) [reply src/services/RecipeService.js]: good catch

File: src/actions/recipeActions.js [hunk 3]
```diff
@@ -16,6 +16,21 @@ const receiveRecipeRecommendations = (recipeRecs) => ({
...
+});
+
+const receiveRecipeToLike = (recipeToLike) => ({
+  type: actionTypes.RECEIVE_RECIPE_TO_LIKE,
+  recipeToLike,
+});
+
+const receiveRecipeToUnlike = (recipeToUnlike) => ({
+  type: actionTypes.RECEIVE_RECIPE_TO_UNLIKE,
+  recipeToUnlike,
```
Comment (mavalos1) [inline src/actions/recipeActions.js:31]: should be named LIKE_RECIPE and UNLIKE_RECIPE

File: src/services/RecipeService.js [hunk 4]
```diff
@@ -23,6 +23,21 @@ class RecipeService {
     const route = `${routes.RECIPE}/${routes.RECS}`;
     return getHelper(route);
   };
+
+  getLikedRecipes = () => {
+    const route = `${routes.RECIPE}/${routes.LIKE}`;
+    return getHelper(route);
+  }
+
+  getRecipeToLike = (recipeId) => {
```
Comment (mavalos1) [inline src/services/RecipeService.js:32]: likeRecipe

File: src/services/RecipeService.js [hunk 5]
```diff
@@ -23,6 +23,21 @@ class RecipeService {
...
+    const route = `${routes.RECIPE}/${routes.LIKE}/${recipeId}`;
+    return postHelper(route);
+  }
+
+  getRecipeToUnlike = (recipeId) => {
```
Comment (mavalos1) [inline src/services/RecipeService.js:37]: unlikeRecipe

File: src/reducers/RecipeReducer.js [hunk 6]
```diff
@@ -40,6 +41,33 @@ const RecipeReducer = createReducer(initialState, {
...
+    isFetching: true,
+  }),
+  [actionTypes.FETCH_RECIPE_TO_LIKE]: (state) => ({
+    ...state,
+    isFetching: true,
+  }),
+  [actionTypes.FETCH_RECIPE_TO_UNLIKE]: (state) => ({
+    ...state,
+    isFetching: true,
+  }),
```
Comment (mavalos1) [inline src/reducers/RecipeReducer.js:55]: I feel these are redundant

File: src/reducers/RecipeReducer.js [hunk 7]
```diff
@@ -40,6 +41,33 @@ const RecipeReducer = createReducer(initialState, {
...
+  [actionTypes.RECEIVE_RECIPE_TO_LIKE]: (state, { recipeDetails }) => ({
+    ...state,
+    isFetching: false,
+    recipeDetails,
+  }),
+  [actionTypes.RECEIVE_RECIPE_TO_UNLIKE]: (state, { recipeDetails }) => ({
+    ...state,
+    isFetching: false,
+    recipeDetails,
+  }),
```
Comment (mavalos1) [inline src/reducers/RecipeReducer.js:70]: we're not necessarily re-fetching the recipe after liking, make this also redundant

Comment (otuffaha) [reply src/reducers/RecipeReducer.js:70]: so you don't want it in the reducer at all?

File: src/reducers/RecipeReducer.js (hunk 6 above)
Comment (otuffaha) [reply src/reducers/RecipeReducer.js:55]: @seanyles

File: src/reducers/RecipeReducer.js (hunk 7 above)
Comment (otuffaha) [reply src/reducers/RecipeReducer.js:70]: @seanyles

**Review by seanyles — APPROVED** [review]

File: src/reducers/RecipeReducer.js (hunk 7 above)
Comment (seanyles) [reply src/reducers/RecipeReducer.js:70]: It doesn't hurt. I'm fine with it

**Pull request merged**